### DISCUSSION TITLE
Using Stylet with AvalonDock

### DISCUSSION THREAD

**Discussion by montyjohn** [canton7/Stylet#366] [General]
Mixing Stylet and AvalonDock sounds like a great platform for new Applications, but I just can't get them to play nicely together, and I'm not sure if the problem is with my code, Stylet or AvalonDock.

New windows can be added to AvalonDock either as a `new Layout` for main windows, or `new Anchorable Layout` for side views.

The new Layout works fine with Stylet, but the new Anchorable Layout throws the error of 'Unable to transform ViewModel name AvalonDock.Layout.LayoutAnchorable into a suitable View name'

I've made a barebones app that replicates the problem here if there's anyone kind enough to have a play to see if they can figure out where the problem is.

https://github.com/montyjohn/StyletAvalonDockTest.git

Thanks all.

**Comment by canton7** [comment]
Not sure on that one. I'm not familiar with AvalonDock, but what you've done looks sensible. The next step is to debug what's being returned by the binding. I'll try and have a look over the weekend

**Comment by canton7** [comment]
Turns out, with the `LayoutAnchorable` case, the DataContext for the binding in the `<DockingManager.LayoutItemTemplate>` is a `ContentControl`, whose `Content` property is the `LayoutAnchorable`. So doing `s:View.Model="{Binding Content.Content}"` works (but of course only for the `LayoutAnchorable`)

I'm afraid I'm not familiar enough with AvalonDock to know why it might be doing that...

**Reply by montyjohn** [reply]
Thanks for looking into it, I note you;ve also commented on the AvalonDock git site so hopefully some AvalonDock gurus will come along soon to see if there's a workaround I can use. Thanks.

**Reply by canton7** [reply]
There are a few ways you could work around this today.

One is to define two DataTemplates, one of which binds to `Content` and the other binds to `Content.Content`, and select the appropriate one using `LayoutItemTemplateSelector`.

Another is to put a converter on that binding which contains something like `value is ContentControl c ? c.Content : value`.
